Baseline Code

jareers opened the pull request:
Initial Blade.net code for review and further updates.

jareers committed:
Baseline Code

Initial Blade.net code for review and further updates.

jareers committed:
Function Name Fixed

An extra scope removed.

ybashir commented on Code.gs:
```diff
@@ -0,0 +1,611 @@
+  // let up = PropertiesService.getUserProperties();
+  // up.deleteProperty(CACHED_CONTACT_LIST);
+  let ccl = getCachedContactList();
+  // ccl['jareers@live.com'] = contactTypes.CONTACT;
```
Should remove dead code

jareers replied:
Done.

ybashir commented on Code.gs:
```diff
@@ -0,0 +1,611 @@
+function handleHourlyProcessSwitchChange(e) {
+  let autoProcess = e.formInput[AUTO_PROCESS];
+
+  if (autoProcess === AUTO_PROCESS) {
```
This could be much cleaner, the if condition just needs to decide whether the trigger should be added or deleted. The rest can be out of the condition

jareers replied:
Thanks for pointing it out. I don't know how I missed that. Fixed.

ybashir commented on Code.gs:
```diff
@@ -0,0 +1,611 @@
+  Logger.log('SF: ' + searchFilter);
+  let allThreads = [];
+
+  while (true) {
```
Google Apps Scripts imposes a limit on the number of seconds a function can run. This infinite loop can potentially exceed that limit.

jareers replied:
Done. Limited to 50 threads at max. Tested and it is the upper limit for our processing in 45 secs time limit by Google.

ybashir reviewed (changes requested):
Added a few comments, mostly minor, 1 major. Otherwise ready to merge.

jareers committed:
Fixed Issues

-Added max 50 thread limit for processing (tested and found it to be the upper limit that can be processed in 45 secs time limit put in by Google)
-Removed dead code
-Trimmed redundant code lines

jareers committed:
Cleaning code
